code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/InputBoxNotificationBuilder.kt

```diff
@@ -0,0 +1,297 @@
+        expandedLayout.setTextViewText(
+            R.id.notification_body_expanded, expandedBodyText
+        )
+
+        // add an input box to capture user feedback if the push template is not from an intent
+        if (pushTemplate.isFromIntent || pushTemplate.inputBoxReceiverName.isNullOrEmpty()) {
+            return notificationBuilder
+        }
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Adding an input box to capture text input. The input box receiver name is ${pushTemplate.inputBoxReceiverName}."
+        )
+        addInputTextAction(
+            context,
+            broadcastReceiverClass,
+            notificationBuilder,
+            channelIdToUse,
+            pushTemplate
+        )
+
+        return notificationBuilder
+    }
+
+    /**
+     * Adds an input text action for the notification.
+     *
+     * @param context the application [Context]
+     * @param broadcastReceiverClass the [BroadcastReceiver] class to use as the broadcast receiver
+     * @param builder the [NotificationCompat.Builder] to attach the action buttons
+     * @param channelId the [String] containing the channel ID to use for the notification
+     * @param pushTemplate the [InputBoxPushTemplate] object containing the input box push template data
+     * button is pressed
+     */
+    private fun addInputTextAction(
+        context: Context,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?,
+        builder: NotificationCompat.Builder,
+        channelId: String,
+        pushTemplate: InputBoxPushTemplate
+    ) {
+        val inputHint =
+            if (!pushTemplate.inputTextHint.isNullOrEmpty()) pushTemplate.inputTextHint else DEFAULT_REPLY_LABEL
+        val remoteInput = pushTemplate.inputBoxReceiverName?.let {
+            androidx.core.app.RemoteInput.Builder(it)
+                .setLabel(inputHint)
+                .build()
+        }
+
+        if (remoteInput == null) {
+            Log.warning(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Failed to create a remote input for the input box action, will not add the action."
+            )
+            return
+        }
+
+        val inputReceivedIntent = createInputReceivedIntent(
+            context,
+            broadcastReceiverClass,
+            channelId,
+            pushTemplate
+        )
+
+        val replyPendingIntent =
+            inputReceivedIntent?.let {
+                PendingIntent.getBroadcast(
+                    context,
+                    pushTemplate.tag.hashCode(),
+                    it,
+                    PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_MUTABLE
+                )
+            }
+
+        val action =
+            NotificationCompat.Action.Builder(
+                null,
+                inputHint, replyPendingIntent
+            )
+                .addRemoteInput(remoteInput)
+                .build()
```

Comment:
Formatting seems to be a but odd here. Does the following not work?
```
 val action =  NotificationCompat.Action.Builder(...
                      ....
```

Reply:
it does look a bit weird. i'll try reformatting with the android studio formatter first then run the spotless one.